gnomad_qc/v3/create_release/prepare_vcf_data_release.py

```diff
@@ -0,0 +1,1019 @@
+            }
+            vcf_info_dict.update(hist_dict)
+
+    # Add in silico annotations to info dict
+    vcf_info_dict["cadd_raw_score"] = t["cadd"]["raw_score"]
+    vcf_info_dict["cadd_phred"] = t["cadd"]["phred"]
+
+    vcf_info_dict["revel_score"] = t["revel"]["revel_score"]
+
+    vcf_info_dict["splice_ai_max_ds"] = t["splice_ai"]["splice_ai_score"]
+    vcf_info_dict["splice_ai_consequence"] = t["splice_ai"]["splice_consequence"]
+
+    vcf_info_dict["primate_ai_score"] = t["primate_ai"]["primate_ai_score"]
+
+    return vcf_info_dict
+
+
+def unfurl_nested_annotations(
+    t: Union[hl.MatrixTable, hl.Table],
+    is_subset: bool = False,
+    gnomad_release: bool = False,
+    entries_to_remove: Set[str] = None,
+) -> [hl.struct, List]:
+    """
+    Create dictionary keyed by the variant annotation labels to be extracted from variant annotation arrays, where the
+    values of the dictionary are Hail Expressions describing how to access the corresponding values.
+
+    :param t: Table/MatrixTable containing the nested variant annotation arrays to be unfurled.
+    :param gnomad_release: Should the gnomAD release frequencies be unfurled.
+    :param is_subset: Is this for the release of a subset.
+    :param entries_to_remove: Frequency entries to remove for vcf_export.
+    :return: Dictionary containing variant annotations and their corresponding values.
+    """
+    freq_entries_to_remove_vcf = set()
+    expr_dict = {}
+
+    prefix = ""
+    if is_subset and gnomad_release:
+        prefix = f"gnomad_"
+
+    # Set variables to locate necessary fields, compute freq index dicts, and compute faf index dict
+    if is_subset and not gnomad_release:
+        freq = "cohort_freq"
+        freq_idx = hl.eval(t.globals[f"cohort_freq_index_dict"])
+    else:
+        popmax = f"{prefix}popmax"
+        faf = f"{prefix}faf"
+        freq = f"{prefix}freq"
+        faf_idx = hl.eval(t.globals[f"{prefix}faf_index_dict"])
+        freq_idx = hl.eval(t.globals[f"{prefix}freq_index_dict"])
+
+    # Unfurl freq index dict
+    # Cycles through each key and index (e.g., k=adj_afr, i=31)
+    logger.info("Unfurling freq data...")
+
+    if is_subset and gnomad_release:
+        prefix = f"gnomad-"
+
+    for k, i in freq_idx.items():
+        # Set combination to key
+        # e.g., set entry of 'afr_adj' to combo
+        combo = k
+        combo_dict = {
+            f"{prefix}AC-{combo}": t[freq][i].AC,
+            f"{prefix}AN-{combo}": t[freq][i].AN,
+            f"{prefix}AF-{combo}": t[freq][i].AF,
+            f"{prefix}nhomalt-{combo}": t[freq][i].homozygote_count,
+        }
+        expr_dict.update(combo_dict)
+
+        if k.split("-")[0] in entries_to_remove:
+            freq_entries_to_remove_vcf.update(combo_dict.keys())
+
+    # Add popmax
+    if not is_subset or (is_subset and gnomad_release):
+        combo_dict = {
+            f"{prefix}popmax": t[popmax].pop,
+            f"{prefix}AC_popmax": t[popmax].AC,
```

Comment:
Because of what is returned by `make_info_dict` when popmax is True: https://github.com/broadinstitute/gnomad_methods/blob/e6f8e5f03cab4e5c36af983e1d81208f2c208ac0/gnomad/utils/vcf.py#L568 However, I could change the common code to use `label_delimiter` instead of `_` in the common code if that would be better